File: 2021_7/inorderTraversal.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        ret = []
        def ios(root):
            if not root:
                return 
            ios(root.left)
            ret.append(root.val)
            ios(root.right)
            # self.ios(root)
        ios(root)
        return ret



        
s = Solution()
root = TreeNode(val=1, right=TreeNode(val=2, left=TreeNode(val=3)))
# root = None
ret = s.inorderTraversal(root)
for i in ret:
    print(i)

File: leetcode_hot100/inorderTraversal.cc
#include "head.h"
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> m_path;
    vector<int> inorderTraversal(TreeNode* root) {
        dfs(root);
        return m_path;
    }

    void dfs(TreeNode *node){
        if(!node) return;
        dfs(node->left);
        m_path.push_back(node->val);
        dfs(node->right);
    }
};